postinstall.js

```diff
@@ -1,10 +1,54 @@
 var childProcess = require('child_process')
 
 if (process.platform === 'darwin') {
-  var result = childProcess.spawnSync('./build_ios_frameworks.sh', [], {
+  var carthageVersionProcessResult = childProcess.spawnSync('carthage', ['version'], {
+    shell: 'bash',
+    stdio: 'pipe'
+  })
+
+  if (carthageVersionProcessResult.status != 0) {
+    // `carthage` not found (probably)
+    console.warn(
+      'Warning: Carthage is required to compile frameworks for iOS backend. You can install it with brew: "brew install carthage". After installation go to ./node_modules/react-native-ble-plx and run ./build_ios_frameworks.sh or reinstall node module.'
+    )
+    process.exit(1)
+  }
+
+  const bleClientManagerDirectory = './ios/BleClientManager'
```

Comment:
Are we sure that when running this script we are always in the same directory as a script itself?

Reply:
You can use
https://nodejs.org/docs/latest/api/modules.html#modules_dirname
to get directory of this file.